contract-examples/contracts/ExampleRewardManager.sol

```diff
@@ -31,3 +34,107 @@ contract ExampleRewardManager is Ownable {
     return rewardManager.areFeeRecipientsAllowed();
   }
 }
+
+contract ExampleRewardManagerTest is DSTest {
+  uint blackholeBalance;
+  ExampleRewardManager exampleReceiveFees;
+  uint exampleBalance;
+
+  function setUp() public {
+    blackholeBalance = BLACKHOLE_ADDRESS.balance;
+  }
+
+  function test_sendFeesToBlackhole() public {
+    blackholeBalance = BLACKHOLE_ADDRESS.balance;
+  }
+
+  function test_checkSendFeesToBlackhole() public {
+    assertGt(BLACKHOLE_ADDRESS.balance, blackholeBalance);
+  }
+
+  function test_doesNotSetRewardAddressBeforeEnabled() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    assertEq(rewardManager.readAllowList(exampleAddress), 0);
+
+    try example.setRewardAddress(exampleAddress) {
+      assertTrue(false, "setRewardAddress should fail");
+    } catch {} // TODO should match on an error to make sure that this is failing in the way that's expected
+  }
+
+  function test_setEnabled() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    assertEq(rewardManager.readAllowList(exampleAddress), 0);
+    rewardManager.setEnabled(exampleAddress);
+    assertEq(rewardManager.readAllowList(exampleAddress), 1);
+  }
+
+  function test_setRewardAddress() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    rewardManager.setEnabled(exampleAddress);
+    example.setRewardAddress(exampleAddress);
+
+    assertEq(example.currentRewardAddress(), exampleAddress);
+  }
+
+  function test_setupReceiveFees() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    rewardManager.setEnabled(exampleAddress);
+    example.setRewardAddress(exampleAddress);
+
+    exampleReceiveFees = example;
+    exampleBalance = exampleAddress.balance;
```

Comment:
we should probably assert getRewardAddress == exampleRewardAddress

Reply:
It's redundant. The test will fail if that assertion does not pass. Testing `getRewardAddress` can happen in its own test (can't remember if it does or not). 

When you run the test suite and see both `should be able to change the reward address` AND `should be able to receive fees` failing, it should be more obvious where to look for the issue. If you only see that `should be able to receive fees` is failing, then you have to look at the exact assertion that's failing (which, unfortunately, isn't very clear, we don't have solidity stack-traces).  

If that test is missing, we can add it in another PR. These tests need some cleanup, regardless.